src/impl/conv_direct.jl

```diff
@@ -53,18 +53,16 @@ function conv_direct!(y::AbstractArray{yT,5}, x::AbstractArray{xT,5},
     width, height, depth = input_size(cdims)
     kernel_w, kernel_h, kernel_d = kernel_size(cdims)
     out_c = channels_out(cdims)
-    pad_w_lo, pad_w_hi, pad_h_lo, pad_h_hi, pad_d_lo, pad_d_hi = padding(cdims)
+    pad_w_lo, _, pad_h_lo, _, pad_d_lo, _ = padding(cdims)
     dil_w, dil_h, dil_d = dilation(cdims)
     stride_w, stride_h, stride_d = stride(cdims)
-    out_width, out_height, out_depth = output_size(cdims)
-    
-    # Create a method that, at compile-time, determines how we're going to index into `w`
-    kproj(k, M, cdims::ConvDims{N,S,P,D,true}) where {N, S, P, D} = k
-    kproj(k, M, cdims::ConvDims{N,S,P,D,false}) where {N, S, P, D} = M - k + 1
-    
+
+    # Create a method that determines how we're going to index into `w`.
+    kproj(k, M, cdims::DenseConvDims) = flipkernel(cdims) ? k : (M - k + 1)
```

Comment:
I don't have the data to back this up, but I kind of thought it would be better if the compiler were able to, at compile time, determine which way `kproj()` should be calculated.... Is it really faster to compare the boolean every time rather than have an inlineable function?

Reply:
This was experimented with in https://github.com/FluxML/NNlib.jl/pull/370#discussion_r780782844, and the conclusion then was that there was no noticeable difference. Whether that's changed after the latest round of commits I'm not sure.

Reply:
GH doesn't jump to the link in the resolved review thread, so here's a screenshot: 
![image](https://user-images.githubusercontent.com/4062143/149067568-606830a8-1eae-4a05-b290-9bd9eb9014ea.png)

Reply:
Latest commit reverts this change